climada/hazard/tc_tracks.py

```diff
@@ -322,6 +420,59 @@ def subset(self, filterdict):
 
         return out
 
+    def get_basins(track):
+
+        track_coordinates = gpd.GeoDataFrame(
+            geometry=gpd.points_from_xy(track.lon, track.lat)
+        )
+        return track_coordinates.sjoin(BASINS_GDF, how="left", predicate="within").basin
+
+    def subset_by_basin(self, origin: bool = False):
+        """Subset all tropical cyclones tracks by basin.
+
+        This function collects for every basin the tracks that crossed them. The resulting dictionary
+        maps each basin's name to a list of tropical cyclones tracks that intersected them.
+
+        Parameters
+        ----------
+        self : TCTtracks object
+            The object instance containing the tropical cyclone data (`self.data`).
+        origin : bool
+            Either True or False. If True, the outputs basin will contain only the tracks that originated there.
+            If False, every track that crossed a basin will be present in the basin.
+
+        Returns
+        -------
+        dict_tc_basins : dict
+            A dictionary where the keys are basin names (e.g., "NA", "EP", "WP", etc.) and the
+            values are instances of the `TCTracks` class: effectively all tracks that intersected or originated
+            in each basin, depending on the argument "origin".
```

Comment:
I think it makes much more sense to return for each basin a TCTrack object, and not a list of xarray datasets.